src/ui/auth/uaa/uaa.go

```diff
@@ -87,13 +91,11 @@ func (u *Auth) PostAuthenticate(user *models.User) error {
 	}
 	if dbUser == nil {
 		return u.OnBoardUser(user)
+	} else {
+		user.UserID = dbUser.UserID
+		user.HasAdminRole = dbUser.HasAdminRole
```

Comment:
Does this line fix the bug ? if dbUser is not nil, how about read the User info from database (similiar to "return dao.OnBoardUser(user)") ? There is no need to call `dao.ChangeUserProfile(*user, "Email", "Realname")` in line 97, unless we want to update UAA user's email and Realname when the UAA user logs in Harbor ?

Reply:
Thanks for the comment.
Correct, the intention here is to try to make sure the user record in DB is update to date with user info in UAA.

Reply:
Per discussison with @reasonerjt, UserID and HasAdminRole fields are defined by Harbor core, not by the different auth providers (e.g. uaa, ldap, ...), it's better to assign these 2 fields in a method of a common parent class (or say the caller of PostAuthenticate). So as to avoid missing these 2 assignment lines in PostAuthenticate() of new auth provider.

Reply:
Will track this enhancement in new ticket and address it in future. Now merge this commit to prepare with the patch release.